### PR TITLE
update fmt and fix warnings that popped up with vs 17.2

### PR DESCRIPTION
**Pull request by shuffle2**
This PR fixes some warnings/errors occurring because of `fmt` after updating to latest VS.

This is expected to not build on windows buildbot for now (buildbot checks for updates every saturday, and i haven't manually updated it). So, we'll start seeing this build break in a few days (and then this PR will work).

Making the PR now in case there needs to be some discussion about the changes.

ping @lioncash because of heavy c++/fmt stuff

as an aside, can we just use `<format>` on msvc? As of VS 17.2 it's officially standard-compliant™️ (https://devblogs.microsoft.com/cppblog/msvcs-stl-completes-stdc20/)
It seems like the API (at least the subset we use) should be basically compatible, it's just annoying because it's in a different namespace.

### CODE REVIEW

File: Externals/fmt/include/fmt/core.h [hunk 1]
```diff
@@ -1702,24 +1702,24 @@ template <typename Context, typename T>
 FMT_CONSTEXPR FMT_INLINE auto make_value(T&& val) -> value<Context> {
   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
 
-  constexpr bool formattable_char =
+  static constexpr bool formattable_char =
```
Comment (AdmiralCurtiss) [inline Externals/fmt/include/fmt/core.h:1705]: `[[maybe_unused]]` instead of the `static` should probably do the trick.

Comment (shuffle2) [reply Externals/fmt/include/fmt/core.h:1705]: unfortunately doesn't seem to work.

Comment (shuffle2) [reply Externals/fmt/include/fmt/core.h:1705]: nvm, it does work, thanks. (was just using regex replace and missed that there's actually two different cases that need to be considered)

**Comment by iwubcode** [comment]
> as an aside, can we just use <format> on msvc? As of VS 17.2 it's officially standard-compliant

I mean you might be able to, but I generally ask _why_?  In a general sense, unless a project is trying to limit includes or doesn't want the hassle of setting up the dependency, [fmt provides additional features](https://stackoverflow.com/questions/63586747/what-are-the-differences-between-libfmt-and-stdformat).  I'd even question if `fmt` is faster, though I have nothing to backup that claim.

More specific to Dolphin, GCC 12.1 just released and still doesn't support `std::format`.  Is there a reason it'd be worthwhile to add more platform differences?

**Comment by shuffle2** [comment]
i dont know if you refer to `fmt` being faster for compile or runtime. I haven't tested either.
The main motivation would just be to move to C++ standardized version.
Dolphin does not use any irreplaceable features of `fmt` / features that don't have equivalence in `std` (afaik). The closest I could find is 1 file that uses `fmt::sprintf`, which it probably shouldn't anyway.

**Comment by iwubcode** [comment]
@shuffle2  - we also use [memory_buffer](https://github.com/dolphin-emu/dolphin/blob/7b8e846d0ab1a95c4070c3d89e38180895e536c6/Source/Core/VideoCommon/XFStructs.cpp#L568)

**Comment by shuffle2** [comment]
memory_buffer doesn't really seem important / lacking an equivalent

**Comment by shuffle2** [comment]
besides `<format>`, open question is about if fa17153ebc83cbc0ae7a1d7430d9509db0c6e0d6 is the correct way to resolve the deprecation.

**Comment by sepalani** [comment]
> i dont know if you refer to `fmt` being faster for compile or runtime. I haven't tested either. The main motivation would just be to move to C++ standardized version. Dolphin does not use any irreplaceable features of `fmt` / features that don't have equivalence in `std` (afaik). The closest I could find is 1 file that uses `fmt::sprintf`, which it probably shouldn't anyway.
> 
> edit: i guess we do use `FMT_COMPILE`, but that's it..

I do use named arguments in [one of my PRs](https://github.com/dolphin-emu/dolphin/pull/10198) which doesn't have an equivalent, AFAICT.

> besides `<format>`, open question is about if fa17153 is the correct way to resolve the deprecation.

I'm pretty sure it is since C++20 compile-time strings checks are enforced by default according to the [documentation](https://fmt.dev/latest/api.html#core-api). That's why `fmt::runtime` needs to be used on runtime strings and `fmt::make_args_checked` is redundant.

Regardless, the code looks good to me.

**Review by lioncash — APPROVED** [review]
Code looks good. Personally I'd opt for continuing to use fmt until `<format>` is supported on all compilers we support and then switch off of it, since there are a few API differences between the two

**Comment by lioncash** [comment]
Seems like the Windows builders need an updated VS version

**Comment by shuffle2** [comment]
> Seems like the Windows builders need an updated VS version

yup, i was just going to wait for the builder to auto-update this weekend

**Pull request merged**